### PR TITLE
LA-9: - graphql hooks

### PR DESCRIPTION
**Pull request by kulak91**
#9 removed

### CODE REVIEW

File: gatsby/helpers/pages-graphql-request/pages-graphql-request.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const pagesGraphQLRequest = (): string => `
+query MyQuery {
+  allMdx(
```
Comment (liza-veis) [inline gatsby/helpers/pages-graphql-request/pages-graphql-request.ts]: Let's use same style as in bsa website and also rename pages to lectures. Don't forget to rename file
```suggestion
const lecturesGraphQLRequest = (): string => `
{
  allLectures(
```

File: src/components/main-page/main-page.tsx [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+type Props = {
+  pageContext: MainPageContext;
+};
+
+const MainPage: React.FC<Props> = ({
+  pageContext: {
+    allMdx: { edges },
+  },
+}) => {
   return (
```
Comment (liza-veis) [inline src/components/main-page/main-page.tsx]: Don't pass edges to this page, extract data from edges in createPage level and pass it, type for context will be like:
```typescript
type MainPageContext = {
  lectures: {}[]
}
```

File: src/types/main-page/main-page-context.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { LectureEdge } from '../graphql/lectures/lectures-query';
+
+interface MainPageContext {
```
Comment (liza-veis) [inline src/types/main-page/main-page-context.ts]:
```suggestion
type MainPageContext {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: gatsby/helpers/pages-graphql-request/pages-graphql-request.ts (hunk 1 above)
Comment (kulak91) [reply gatsby/helpers/pages-graphql-request/pages-graphql-request.ts]: "allMdx" is a command, which tells  graphql to get all .mdx extensions, which are visible due to "gatsby-source-filesystem" module in gatsby-config.ts.
However, renamig approved

File: gatsby/create-main-page/create-main-page.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  graphql: CreatePagesArgs['graphql'],
+  reporter: Reporter,
+): Promise<void> => {
+  const pages = await graphql(lecturesGraphQLRequest());
+
+  if (pages.errors && !pages.data) {
+    reporter.panicOnBuild('Error while running GraphQL query.');
+  }
+
+  const mainPageContext = pages.data as LecturesQuery;
```
Comment (liza-veis) [inline gatsby/create-main-page/create-main-page.ts:19]: Here we need to map all the data and pass it in format: 

```typescript
[{
    slug,
    author,
    description,
    duration,
    publishedAt,
    title,
}]
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by liza-veis — APPROVED** [review]
👍

**Pull request merged**